app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelTest.kt

```diff
@@ -0,0 +1,93 @@
+import org.amshove.kluent.internal.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(CoroutineTestExtension::class)
+class NewConversationViewModelTest {
+
+    @Test
+    fun `when search with search query, return results for known and public search`() {
+        runTest {
+            // Given
+            val (_, viewModel) = NewConversationViewModelArrangement().arrange()
+
+            // When
+            viewModel.search("search")
+            advanceTimeBy(501) // 500ms debounce
+
+            // Then
+            assertEquals(
+                viewModel.state.localContactSearchResult.searchResultState, SearchResultState.Success(
+                    result = listOf(
+                        Contact(
+                            id = "knownValue",
+                            domain = "domain",
+                            name = "knownUsername",
+                            avatarData = UserAvatarData(
+                                asset = ImageAsset.UserAvatarAsset(userAssetId = UserAssetId("value", "domain")),
+                                availabilityStatus = UserAvailabilityStatus.NONE
+                            ),
+                            label = "knownHandle",
+                            connectionState = ConnectionState.NOT_CONNECTED
+                        )
+                    )
+                )
+            )
+            assertEquals(
+                viewModel.state.publicContactsSearchResult.searchResultState, SearchResultState.Success(
+                    result = listOf(
+                        Contact(
+                            id = "publicValue",
+                            domain = "domain",
+                            name = "publicUsername",
+                            avatarData = UserAvatarData(
+                                asset = ImageAsset.UserAvatarAsset(userAssetId = UserAssetId("value", "domain")),
+                                availabilityStatus = UserAvailabilityStatus.NONE
+                            ),
+                            label = "publicHandle",
+                            connectionState = ConnectionState.NOT_CONNECTED
+                        )
+                    )
+                )
+            )
+        }
+    }
+
+    @Test
+    fun `when search with search query, return failure for known and public search`() {
+        runTest {
+            // Given
+            val (_, viewModel) = NewConversationViewModelArrangement()
+                .withFailureKnownSearchResponse()
+                .withFailurePublicSearchResponse()
+                .arrange()
+
+            // When
+            viewModel.search("search")
+            advanceTimeBy(501) // 500ms debounce
+
+            // Then
+            assertEquals(
+                viewModel.state.localContactSearchResult.searchResultState, SearchResultState.Failure(2131755295)
```

Comment:
in the mock Result.Failure.InvalidRequest but here it's SearchResultState.Failure so I assume that `2131755295` is hashcode for InvalidRequest, in ViewModel we handle all failures exactly the same

Reply:
I think, we could do something like: `viewModel.state.localContactSearchResult.searchResultState::class` and compare it to `Result.Failure.InvalidRequest::class` or better, maybe this is possible also:

`assertTrue(viewModel.state.localContactSearchResult.searchResultState is Result.Failure.InvalidRequest)`

Hash codes can vary from jvm to jvm, leading to flakiness.